src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -1870,7 +2105,7 @@ public void ReplaceIndexes(string oldIndexName, string replacementIndexName, Can
                                 {
                                     using (IndexLock(newIndex.Name))
                                     {
-                                        newIndex.Stop(disableIndex: true);
+                                        newIndex.Stop(disableIndex: false);
```

Comment:
Why we switched here?

Reply:
this is core code for this variable:
```
if (disableIndex)
{
    lock (_disablingIndexLock)
    {
        _indexDisabled = true;
        _mre.Set(ignoreThrottling: true);
    }
}
else
{
    _indexingProcessCancellationTokenSource?.Cancel();
}
```

Upon replacing it looks reasonable to me to cancel the thread rather than to disable it.
Also, for pending indexes I _have_ to cancel it, since I'm waiting on a different MRE